sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/catalog/ExternalCatalogSuite.scala

```diff
@@ -778,6 +778,24 @@ abstract class ExternalCatalogSuite extends SparkFunSuite with BeforeAndAfterEac
       "db2", "tbl1", Seq(part1.spec), ignoreIfNotExists = false, purge = false, retainData = false)
     assert(fs.exists(partPath))
   }
+
+  test("drop database cascade with function defined") {
```

Comment:
@gatorsmile I took another look at the Hive issue, it should be a bug caused by hive's `FunctionRegistry`, so I think spark is not affected, since we have already get rid of those non-native commands. Thank you for your patience!